impl/parquet_reader.go

```diff
@@ -70,19 +73,50 @@ func (p *ParquetReader) GetNumberOfRows() int64 {
 }
 
 func toKafkaMessage(message ParquetMessage) (*kafka.Message, error) {
+	timestamp, err := time.Parse(time.RFC3339, message.Timestamp)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to convert string to time.Time: %s", message.Timestamp)
+	}
+
 	var headers []kafka.Header
 	if len(message.Headers) > 0 {
 		err := json.Unmarshal([]byte(message.Headers), &headers)
 		if err != nil {
 			return nil, errors.Wrapf(err, "Failed to unmarshal kafka headers: %s", message.Headers)
 		}
 	}
-	return &kafka.Message{
+
+	var timestampType int
+	switch message.TimestampType {
+	case kafka.TimestampCreateTime.String():
+		timestampType = int(kafka.TimestampCreateTime)
+	case kafka.TimestampLogAppendTime.String():
+		timestampType = int(kafka.TimestampLogAppendTime)
+	case kafka.TimestampNotAvailable.String():
+		fallthrough
+	default:
+		timestampType = int(kafka.TimestampNotAvailable)
+	}
+
+	kafkaMessage := &kafka.Message{
 		Value: []byte(message.Value),
 		TopicPartition: kafka.TopicPartition{
 			Topic: &message.Topic,
 		},
-		Key:     []byte(message.Key),
-		Headers: headers,
-	}, nil
+		Key:           []byte(message.Key),
+		Headers:       headers,
+		Timestamp:     timestamp,
+		TimestampType: kafka.TimestampType(timestampType),
+	}
+
+	if message.Offset != nil {
+		offset, err := strconv.Atoi(*message.Offset)
```

Comment:
I'm considering why we need to convert to a string offset when writing and parse as int64 when reading, instead of directly saving it as int64?

Reply:
Offset is not a int64, its a custom type in library, and in the library we have a function to convert it into string. Thats why storing it as string and converting string to int because kafka.Offset() takes int as input
```
if message.Offset != nil {
		offset, err := strconv.Atoi(*message.Offset)
		if err != nil {
			return nil, errors.Wrapf(err, "Failed to convert string to int for message offset: %s", message.Offset)
		}
		kafkaMessage.TopicPartition.Offset = kafka.Offset(offset)
		kafkaMessage.TopicPartition.Partition = *message.Partition
	}
```

Reply:
You're right.
You're right, but Kafka's custom type is, in fact, int64.

```go
// Offset type (int64) with support for canonical names
type Offset int64
```

Reply:
yes, but we have to either convert it during writing to file or writing to kafka.